pydantic_redis/abstract.py

```diff
@@ -21,27 +21,54 @@ class Config:
         arbitrary_types_allowed = True
         orm_mode = True
 
-def pydantic_serializer(data):
-    if isinstance(data, _AbstractModel):
-        return data.json()
-
 class _AbstractModel(BaseModel):
     """
     An abstract class to help with typings for Model class
     """
     _store: _AbstractStore
     _primary_key_field: str
 
+    @classmethod
```

Comment:
So here is what I seem to be getting from the code:
- When inserting, you check it it is a list of models, and if so, you insert each of these, and replace them with a list of their primary keys
- When selecting, you check if the given field is has annotation like `List[Model]`, and if so, you eagerly load those using their primary keys.

The methodology makes sense.

Reply:
A few recommendations:

- Consider transfering the `_serialize_models_list` logic to the `__get_serializable_dict` method of `Model` in the model.py module. There is already a check for if the data is of instance `Model`. You could do something similar but for `List`. There might be some recursion needed to pull it off.

```python
  @classmethod
    def __get_serializable_dict(cls,
                                pipeline: Pipeline,
                                record: Union[_AbstractModel, Dict[str, Any]],
                                life_span: Optional[Union[float, int]] = None) -> Dict[str, Any]:
        data = record.items() if isinstance(record, dict) else record
        new_data = {}

        for k, v in data:
            key, value = k, v

            if isinstance(v, list):
                key = f"____{key}" # or some easy to see prefix...as long as it does not collide with the one for `Model`
                # __get_serializable_list then checks if it is a list of Models or not 
                value = __get_serializable_list(pipeline=pipeline, record=v, life_span=life_span)

            if isinstance(v, Model):
                key = f"__{key}"
                value = v.__class__.__insert_on_pipeline(
                    _id=None, pipeline=pipeline, record=v, life_span=life_span)

            new_data[key] = value
        return new_data
```

- Consider transfering the `_deserialize_models_list` logic to the `__parse_dict_list` method of the `Model` in model.py module. In that method, you might find an easier method to check a field's type annotations `field_types = typing.get_type_hints(cls)`
    - See this line: `for k in parsed_data[0].keys() if k.startswith("__")`? I added "__" prefixes to any key that had a `Model` as an annotation such that redis has them saved as say `__author` instead of `author`. You could do something similar for `List[Model]` fields.

```python
 @classmethod
    def __parse_dict_list(cls, data: List[Dict[bytes, Any]]) -> List[Dict[str, Any]]:
        parsed_data = [cls.deserialize_partially(record) for record in data if record != {}]
        if len(parsed_data) > 0:
            field_types = typing.get_type_hints(cls)
            nested_model_map: Dict[str, typing.Type[Model]] = {
                k: field_types.get(k.lstrip("__"))
                for k in parsed_data[0].keys() if k.startswith("__")
            }
```
    
Not sure if that makes sense.

Reply:
The `test_update_nested_list_of_models` was also failing I guess because you are still working on it.